doc/application/index.rst

```diff
@@ -1446,6 +1446,156 @@ After running the preprocessor, the final device tree used in the
+child image together so that they can be easily flashed onto the device through
+the "ninja flash" target.  Meaning, for the simplest scenario one can
+enable and integrate an additional image without even realizing it with the
+same workflow as is used for additional libraries, enablement through Kconfig.
+
+Configuring child images is done with the same mechanisms as for the
+parent images. Through CMake variables, Kconfig input fragments, and
+modifying the .config file in the build directory. For instance, to
+one could run ``cmake -Dmcuboot_CONF_FILE=prj_a.conf
+-DCONF_FILE=app_prj.conf``. To change the CONF_FILE for the mcuboot
+image and the parent image. As seen in CONF_FILE, all CMake Cache
+variables that are image-specific are given a prefix to disambiguate
+them. To simplify configuation of single-image builds the toplevel
+image has the empty string as it's prefix.
+
+The same prefix convention is used to disambiguate targets. This means
+that for instance to run menuconfig one would invoke the
+``menuconfig`` target to configure the toplevel image and
+``mcuboot_menuconfig`` to configure the MCUBoot child image.
+
+Defining new child images
+=========================
+
+This section describes how to take an existing parent image and turn
+it into a child image that can be enabled. A parent image typically
+consists of source code, Kconfig fragments, and build script code. The
+source code and Kconfig fragments can be re-used as-is, but several
+changes to the build scripts are necessary.
+
+As mentioned earlier, each target needs a prefix to disambiguate. This
+includes the library target 'app', so any references to 'app' would
+need to be changed to '${IMAGE}app'. Or, even better, one would use
+the zephyr_library_ API instead of the target_* API to indirectly
+modify '${IMAGE}app'.
+
+After the application build scripts have been ported we can write some
+build scripts as shown below to connect the build scripts of the
+parent and child. This code should be placed somewhere in-tree that is
+conditional on a Kconfig option for having the parent image use the
+child image.
+
+.. code-block:: cmake
+
+  set(child_image_name mcuboot)
+  zephyr_add_executable(${child_image_name} require_build)
+
+  set(build_directory ${CMAKE_CURRENT_BINARY_DIR}/mcuboot
+  set(child_image_application_directory ${MCUBOOT_BASE}/boot/zephyr)
+
+  if(${require_build})
+    add_subdirectory(${child_image_application_directory} ${build_directory})
+  endif()
+
+In the above code 'zephyr_add_executable' registers the child image as
+present in the build, and 'add_subdirectory' actually executes the
+child's build scripts. Note that in addition the child image's
+application build scripts being executed, most of the core build
+scripts are executed for a second time as well, but now with a
+different Kconfig'uration and possibly DeviceTree settings.
+
+Some Kconfig options must be added to allow the parent image to choose how to
```

Comment:
I'd recommend writing this as:
```
Some Kconfig options must be added to allow the parent image to choose how to
include the child image. The three options are:

1. **Build from source**: this results in the child image being built
   from source alongside the parent image,

2. **Include as hex file**: this requires a prebuilt hex file to be specified
   in Kconfig, which will be merged with the parent image hex file.

3. **Skip building entirely**:
   this will result in the child image not being built.

These options are shown below for MCUBoot:
```